core/application/impl/app_state_manager_impl.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_APPLICATION_DISPATCHER_IMPL
+#define KAGOME_APPLICATION_DISPATCHER_IMPL
+
+#include "application/app_state_manager.hpp"
+
+#include <condition_variable>
+#include <csignal>
+#include <mutex>
+#include <queue>
+
+#include "common/logger.hpp"
+
+namespace kagome {
+
+  class AppStateManagerImpl : public AppStateManager {
+   public:
+    AppStateManagerImpl();
+    AppStateManagerImpl(const AppStateManagerImpl &) = delete;
+    AppStateManagerImpl(AppStateManagerImpl &&) noexcept = delete;
+
+    ~AppStateManagerImpl() override;
+
+    AppStateManagerImpl &operator=(AppStateManagerImpl const &) = delete;
+    AppStateManagerImpl &operator=(AppStateManagerImpl &&) noexcept = delete;
+
+    void atPrepare(Callback &&cb) override;
+    void atLaunch(Callback &&cb) override;
+    void atShutdown(Callback &&cb) override;
+
+    void run() override;
+    void shutdown() override;
+
+    State state() const override {
```

Comment:
```suggestion
    inline State state() const override {
```

Reply:
Methods defined in class declaration is inline defaultly.